src/app/scripts/cac/map/cac-map-control.js

```diff
@@ -644,6 +748,21 @@ CAC.Map.Control = (function ($, Handlebars, cartodb, L, turf, _) {
         itineraries = {};
     }
 
+    /**
+     * Remove all itinerary map layers by inspecting for the GeoJSON `from` property.
+     *
+     * TODO: find better way to handle potential race conditions with live updating,
+     * which mean itinerary layers cannot be reliably removed by using the dictionary of
+     * itineraries stored in the map control during or on end of waypoint drag.
+     */
+    function removeItineraryLayers() {
+        map.eachLayer(function(layer) {
+            if (layer.feature && layer.feature.properties && layer.feature.properties.from) {
+                map.removeLayer(layer);
+            }
+        });
+    }
```

Comment:
What about maintaining an `itineraryLayerList` array that you push layers itinerary layers onto whenever you add them to the map?  Then this could loop over that and remove each layer then set the array back to `[]`.  Leaflet doesn't appear to mind if you call removeLayer on a layer that was already removed.

Pro: none of this "if it has a `from` attribute, ice it" business.
Con: have to be sure to push itinerary layers to the array when adding them to the map (currently done in 2 places). Possibly a wrapper function would be prudent.  Also probably want to remove them from the array in `updateItineraryLayer`.